### PR TITLE
[iOS] Enable skipped System.Threading tests

### LINKED ISSUE #51400: System.Threading.Tests fails on iOS/tvOS
System.Threading.Tests
System.Threading.Tests.AutoResetEventTests.PingPong
System.Threading.Tests.ManualResetEventTests.PingPong
System.Threading.Tests.SemaphoreTests.AnonymousProducerConsumer
System.Threading.Tests.ReaderWriterLockTests.WaitingWritersTest
System.Threading.Tests.ReaderWriterLockTests.WaitingUpgradersTest
System.Threading.Tests.ReaderWriterLockTests.BasicLockTest
System.Threading.Tests.ReaderWriterLockTests.ReadersWaitingOnWaitingUpgraderTest
System.Threading.Tests.ReaderWriterLockTests.ReadersWaitingOnWaitingWriterTest
System.Threading.Tests.ReaderWriterLockTests.AtomicRecursiveReaderTest
System.Threading.Tests.ReaderWriterLockTests.AtomicDowngradeTest
System.Threading.Tests.ExecutionContextTests.FlowTest
System.Threading.Tests.ExecutionContextTests.CaptureThenSuppressThenRunFlowTest

```
System.Threading.Tests.dll   Failed: 12

Test collection for System.Threading.Tests.dll
System.Threading.Tests.AutoResetEventTests.PingPong
    System.AggregateException : One or more errors occurred. (Assert.True() Failure\nExpected: True\nActual:   False) (Assert.True() Failure\nExpected: True\nActual:   False)\n---- Assert.True() Failure\nExpected: True\nActual:   False\n---- Assert.True() Failure\nExpected: True\nActual:   False
System.Threading.Tests.ManualResetEventTests.PingPong
    System.AggregateException : One or more errors occurred. (Assert.True() Failure\nExpected: True\nActual:   False) (Assert.True() Failure\nExpected: True\nActual:   False)\n---- Assert.True() Failure\nExpected: True\nActual:   False\n---- Assert.True() Failure\nExpected: True\nActual:   False
System.Threading.Tests.SemaphoreTests.AnonymousProducerConsumer
    System.AggregateException : One or more errors occurred. (Assert.True() Failure\nExpected: True\nActual:   False)\n---- Assert.True() Failure\nExpected: True\nActual:   False
System.Threading.Tests.ReaderWriterLockTests.WaitingWritersTest
    Assert.InRange() Failure\r\nRange:  (0 - 60000)\r\nActual: 60001
System.Threading.Tests.ReaderWriterLockTests.WaitingUpgradersTest
    Assert.True() Failure\nExpected: True\nActual:   False
System.Threading.Tests.ReaderWriterLockTests.BasicLockTest
    Assert.True() Failure\nExpected: True\nActual:   False
System.Threading.Tests.ReaderWriterLockTests.ReadersWaitingOnWaitingUpgraderTest
    Assert.InRange() Failure\r\nRange:  (0 - 60000)\r\nActual: 60001
System.Threading.Tests.ReaderWriterLockTests.ReadersWaitingOnWaitingWriterTest
    Assert.InRange() Failure\r\nRange:  (0 - 60000)\r\nActual: 60001
System.Threading.Tests.ReaderWriterLockTests.AtomicRecursiveReaderTest
    Assert.InRange() Failure\r\nRange:  (0 - 60000)\r\nActual: 60001
System.Threading.Tests.ReaderWriterLockTests.AtomicDowngradeTest
    Assert.InRange() Failure\r\nRange:  (0 - 60000)\r\nActual: 60001
System.Threading.Tests.ExecutionContextTests.FlowTest
    System.AggregateException : One or more errors occurred. (Assert.True() Failure\nExpected: True\nActual:   False)\n---- Assert.True() Failure\nExpected: True\nActual:   False
System.Threading.Tests.ExecutionContextTests.CaptureThenSuppressThenRunFlowTest
    System.AggregateException : One or more errors occurred. (Assert.True() Failure\nExpected: True\nActual:   False)\n---- Assert.True() Failure\nExpected: True\nActual:   False

### PR DESCRIPTION
**Pull request by steveisok**
Fixes https://github.com/dotnet/runtime/issues/51400

### CODE REVIEW

**Review by lewing — APPROVED** [review]
LGTM assuming it is green, did you mean to include the extra newline?

File: src/libraries/Microsoft.Extensions.Configuration/tests/FunctionalTests/ArrayTests.cs [hunk 1]
```diff
@@ -47,7 +47,6 @@ public class ArrayTests : IDisposable
 ";
 
         [Fact]
-        [ActiveIssue("https://github.com/dotnet/runtime/issues/60583", TestPlatforms.iOS | TestPlatforms.tvOS)]
         public void DifferentConfigSources_Merged_KeysAreSorted()
```
Comment (mdh1418) [inline src/libraries/Microsoft.Extensions.Configuration/tests/FunctionalTests/ArrayTests.cs:50]: This one needs to be disabled again based off of the comment that it still fails https://github.com/dotnet/runtime/issues/60583#issuecomment-1206831713

**Comment by mdh1418** [comment]
Looks like all of the tests in the issue now pass on tvOS device. There was no xml produced but the `net.dot.System.Threading.Tests.log` contained all of the test results.

**Pull request merged**